src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3978,14 +3985,199 @@ bool   os::win32::_is_windows_server         = false;
 // including the latest one (as of this writing - Windows Server 2012 R2)
 bool   os::win32::_has_exit_bug              = true;
 
+int    os::win32::_major_version             = 0;
+int    os::win32::_minor_version             = 0;
+int    os::win32::_build_number              = 0;
+int    os::win32::_build_minor               = 0;
+
+void os::win32::initialize_windows_version() {
+  if (_major_version > 0) {
+    return; // nothing to do if the version has already been set
+  }
+
+  VS_FIXEDFILEINFO *file_info;
+  TCHAR kernel32_path[MAX_PATH];
+  UINT len, ret;
+
+  // Get the full path to \Windows\System32\kernel32.dll and use that for
+  // determining what version of Windows we're running on.
+  len = MAX_PATH - (UINT)strlen("\\kernel32.dll") - 1;
+  ret = GetSystemDirectory(kernel32_path, len);
+  if (ret == 0 || ret > len) {
+    warning("GetSystemDirectory() failed: GetLastError->%ld.", GetLastError());
```

Comment:
Sorry if it was unclear but all warnings should ideally use `os::last_error()` to report a user-friendly error message.